text/0069-api-v2.md

```diff
@@ -2,216 +2,237 @@
 
-- Reject all requests with `api_version=2` in the context.
+- Reject all requests with `api-version=2` in the context.
 
-#### TiKV Client
+### TiKV Client
 
 Provide two modes for users:
 
 - V2:
-  - Fetch keyspace prefix by keyspace name from PD and then prepend `k{keyspace prefix}x` on TxnKV keys or prepend `k{keyspace prefix}r` on RawKV keys.
-  - Set `api_version=2` in TiKV gRPC's `Context`.
-  - Disallow specify CF in `RawClient`.
-  - Allow user to specify a keyspace for a session of `RawClient` or `TxnCient`. Default keyspace is named `default`.
-  - Fetch keyspace information from PD every 5 mins. Destory the client session if the keyspace it's using is deleted.
+  - Prepend `x{keyspace}` before TxnKV keys or prepend `r{keyspace}` before RawKV keys.
+  - `Keyspace` is optional and defaults to `0` for backward compatible
+  - Set `api_version=2` in `kvrpcpb.Context`.
+  - Disallow specifying `cf` in `RawClient`.
 
 - V1:
-  - Behaves jusk like current client.
-  - Set `api_version=1` in TiKV gRPC's `Context`.
+  - Set `api_version=1` in `kvrpcpb.Context`.
+  - Besides above, behaves just like current client.
 
 Listed below is the compatibility matrix:
 
-|              | V1 Server | V2 Server |
-| ------------ | --------- | --------- |
-| V1 RawClient | Raw Data  | Forbidden |
-| V1 TxnClient | Txn Data  | TiDB Data |
-| V2 RawClient | Forbidden | Raw Data  |
-| V2 TxnClient | Forbidden | Txn Data  |
+|                       | V1 Server | V1TTL Server | V2 Server |
+| --------------------- | --------- | ------------ | --------- |
+| V1 RawClient          | Raw       | Raw          | Error     |
+| V1 RawClient with TTL | Error     | Raw          | Error     |
+| V1 TxnClient          | Txn       | Error        | Error     |
+| V1 TiDB               | TiDB Data | Error        | TiDB Data |
+| V2 RawClient          | Error     | Error        | Raw       |
+| V2 TxnClient          | Error     | Error        | Txn       |
+
+### Garbage Collection
+
+Garbage Collection (`GC` for short) is necessary for recycling the space from:
+
+1. Obsolete versions of [MVCC]
+2. TTL expired entires
+3. Deleted entries
+
+There are two garbage collector existing in TiKV. One for transactional data, which utilizes [Compaction Filter] of RocksDB to drop obsolete versions during compaction, and schedules background tasks to physically remove deleted entires. Another one for RawKV data with TTL, which also utilizes [Compaction Filter] to drop TTL expired entries.
+
+The **obsolete versions** are those older than a TSO called `GC Safe Point`. `GC Safe Point` is advanced by TiDB GC leader, which will keep versions for a life time defaults to 10 minutes. At the same time, services such as [TiDB BR] & [TiCDC] can setup a `Service Safe Point` preventing `GC Safe Point` to advance, and keep more versions to meet service requirement (e.g, [TiDB BR] requires to keep all versions after `backup timestamp` for snapshot consistence).
```

Comment:
The lastet version of one key should not be called the `obsolete versions` even of it's ts is older than `GC Safe Point`.

Reply:
Fixed, PTAL~